source/Calamari/Serialization/InheritedClassConverter.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Calamari.Serialization
+{
+    public abstract class InheritedClassConverter<TBaseResource, TEnumType> : JsonConverter
```

Comment:
Taken from the server, however, existing contract resolver will be used. We are dealing with a nested object serialized into the property for file selections for example.